File: Reto # -14/Python/Fechas.py
from datetime import datetime

## DeathbatO
# Función para obtener la fecha actual
def get_current_date():
    return datetime.now()

# Función para calcular la edad
def calculate_age(birth_date, current_date):
    age = current_date.year - birth_date.year
    if (current_date.month, current_date.day) < (birth_date.month, birth_date.day):
        age -= 1
    return age

def main():
    # Obtener la fecha actual
    current_date = get_current_date()

    # Especificar tu fecha de nacimiento
    birth_date = datetime(1990, 4, 1)  # Por ejemplo, 1 de abril de 1990

    # Calcular la edad
    age = calculate_age(birth_date, current_date)

    # Imprimir el resultado
    print(f"Han transcurrido {age} años desde tu nacimiento.")

if __name__ == "__main__":
    main()

File: Reto # -14/Java/Fechas.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.Period;

//*
*    @author DeathbatO
*/

public class Fechas{
    public static void main(String[] args) {
        // Obtener la fecha actual
        LocalDateTime fechaActual = LocalDateTime.now();

        // Especificar tu fecha de nacimiento
        LocalDateTime fechaNacimiento = LocalDateTime.of(1990, 4, 1, 0, 0, 0); // Por ejemplo, 1 de abril de 1990

        // Calcular la diferencia entre las dos fechas
        Period periodo = Period.between(LocalDate.from(fechaNacimiento), LocalDate.from(fechaActual));
        int añosTranscurridos = periodo.getYears();

        // Imprimir el resultado
        System.out.println("Han transcurrido " + añosTranscurridos + " años desde tu nacimiento.");
    }
}
